docs/modules/ROOT/pages/push.adoc

```diff
@@ -0,0 +1,94 @@
+
+=== Installation
+
+For a Cordova App two packages need to be installed: `@aerogear/push` and the platform push plugin:
+
+[source,bash]
+----
+$ npm install --save @aerogear/push
+$ cordova plugin add phonegap-plugin-push
+----
+
+Ionic Apps require an additional dependency, the link:https://ionicframework.com/docs/native/push/[Ionic Native Push Library]
+
+[source,bash]
+----
+npm install --save @ionic-native/push
+----
+
+=== Configuration
+
+Make sure that your `mobile-serivces.json` file has a configuration object for push. If not you can retrieve it from the mobile client UI in Openshift or using the mobile cli:
+
+[source,bash]
+----
+$ mobile get clientconfig <mobile client id> --namespace=<namespace> -o json
+----
+
+Copy the over the `push` part of the configuration:
+
+----
+{
+  ...
+  "services": [
+    {
+      "id": "push",
+      "name": "push",
+      "type": "push",
+      "url": <backend url>,
+      "config": {
+        "ios":      <iOS specific config>
+        "android":  <Android specific config>
+      }
+    }
+  ]
+}
+----
+
+Next you need to configure the push provider itself. `phonegap-plugin-push` defaults to Firebase Cloud Messaging on Android and iOS. Login to your link:https://firebase.google.com/[Firebase account] and retrieve the config file (`google-services.json` or `google-services.plist`). Include it in the root of your project folder. Please refer to the link:https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/INSTALLATION.md#android-details[Plugin documentation] for more details.
+
+=== Usage
+
+To make use of the Aerogear Unifiedpush Server, devices need to be registered with it. This in handled by `@aerogear/push` and should be performed after the platform push plugin itself has finished initialization:
+
+[source,typescript]
+----
+import { Push, PushObject } from "@ionic-native/push";
+import { PushRegistration } from "@aerogear/push";
+
+const pushObject = new Push().init({
+    android: {},
+    ios: {
+      alert: true,
+      badge: true,
+      sound: true
+    }
+});
+
+pushObject.on('registration').subscribe(data => {
+  // PushRegistration will handle the device registration with the Unifiedpush Server
+  new PushRegistration().register(data.registrationId).then(() => {
+    // Registration with UPS successful, you can now send push notifications from the UPS UI
+  }).catch(err => {
+    // Error on device registration
+  });
+});
+----
+
+==== Options
+
+You can specify a number of options for the registration. The signature of `PushRegistration#register` is:
```

Comment:
-1. Intelisense, api docs can be used here. This can outdated etc.

Reply:
Actually let's leave it. You put hard work to get that done so no sense to remove it.

Reply:
I don't mind removing the code snippet, just wanted to explain the parameters. But maybe they should be documented in the UPS docs instead?

Reply:
We have something here: https://github.com/aerogear/aerogear-js-sdk/blob/master/packages/push/src/PushRegistration.ts#L34-L40